subsys/bluetooth/host/id.c

```diff
@@ -1368,6 +1364,32 @@ int bt_id_delete(uint8_t id)
 	return 0;
 }
 
+
+int bt_id_delete(uint8_t id)
+{
+	if (id == BT_ID_DEFAULT || id >= bt_dev.id_count) {
+		return -EINVAL;
+	}
+
+	return id_delete(id);
+}
+
+int bt_id_delete_all(void)
+{
+	for (uint8_t id = 0U; id < bt_dev.id_count; id++) {
```

Comment:
shouldn't we cache `bt_dev.id_count`? It seems it is decremented in `id_delete()` so I'm worried the flow of this for-loop might be erratic.

Reply:
Yeah that's a good point